Sources/NIOSOCKS/Channel Handlers/SOCKSClientHandler.swift

```diff
@@ -79,21 +79,25 @@ public final class SOCKSClientHandler: ChannelDuplexHandler {
     }
     
     public func write(context: ChannelHandlerContext, data: NIOAny, promise: EventLoopPromise<Void>?) {
-        guard self.state.proxyEstablished else {
-            self.bufferedWrites.append((data, promise))
-            return
-        }
-        self.writeBufferedData(context: context)
-        context.write(data, promise: promise)
+        let buffer = self.unwrapOutboundIn(data)
+        self.bufferedWrites.append((data, promise))
```

Comment:
This adds overhead into the write path after the negotiation has completed. Rather than unconditionally buffer this, can we keep the fast-path here?

Reply:
Yeh good idea